src/Metadata/Processor/IndexProcessor.php

```diff
@@ -0,0 +1,49 @@
+<?php declare(strict_types=1);
+
+namespace Fazland\ODM\Elastica\Metadata\Processor;
+
+use Fazland\ODM\Elastica\Annotation\Index;
+use Fazland\ODM\Elastica\Metadata\DocumentMetadata;
+use Kcs\Metadata\Loader\Processor\ProcessorInterface;
+use Kcs\Metadata\MetadataInterface;
+
+class IndexProcessor implements ProcessorInterface
+{
+    /**
+     * {@inheritdoc}
+     *
+     * @param DocumentMetadata $metadata
+     * @param Index            $subject
+     */
+    public function process(MetadataInterface $metadata, $subject): void
+    {
+        $analysis = [
+            'filter' => [],
+            'tokenizer' => [],
+            'analyzer' => [],
+        ];
+
+        foreach ($subject->filters ?? [] as $filter) {
```

Comment:
The annotation object (Index in this case) is instantiated from the annotation library. If the option is not defined it will be set as `null`. The properties must be public and nullable to comply with the annotations library.

Reply:
again, that's so bad.. 😲